core/web/apiv2/auth.py

```diff
@@ -16,10 +22,33 @@
 ALGORITHM = yeti_config.get('auth', "algorithm")
 YETI_AUTH = yeti_config.get('auth', "enabled")
 
+AUTH_MODULE = yeti_config.get('auth', "module")
+if AUTH_MODULE == 'oidc':
+    if not yeti_config.get('auth', "oidc_client_id") or \
+       not yeti_config.get('auth', "oidc_client_secret") or \
+       not yeti_config.get('auth', "oidc_discovery_url"):
+        raise Exception("OIDC AUTHENTICATION requires OIDC_CLIENT_ID, OIDC_CLIENT_SECRET, and OIDC_DISCOVERY_URL to be set in the configuration file")
+
 oauth2_scheme = OAuth2PasswordBearer(tokenUrl="api/v2/auth/token", auto_error=False)
 cookie_scheme = APIKeyCookie(name="yeti_session", auto_error=False)
 api_key_header = APIKeyHeader(name="x-yeti-apikey")
 
+def get_oauth_client() -> OAuth:
+    client_id = yeti_config.get('auth', "oidc_client_id")
+    client_secret = yeti_config.get('auth', "oidc_client_secret")
+    discovery_url = yeti_config.get('auth', "oidc_discovery_url")
+
+    client = OAuth()
+    client.register(
+        name='oidc',
+        server_metadata_url=discovery_url,
+        client_kwargs={
+            'scope': 'openid email profile',
+        },
+        client_id=client_id,
+        client_secret=client_secret,
+    )
+    return client
```

Comment:
Do you want to let calls to this method handle exceptions? No fields from the configuration are checked (at least not empty).

Reply:
I missed the check at import time... 🤦